packages/react-native/src/Authenticator/common/FederatedProviderButtons/types.ts

```diff
@@ -1,16 +1,35 @@
-import { TextStyle, ViewStyle } from 'react-native';
+import { StyleProp, TextStyle, ViewProps, ViewStyle } from 'react-native';
 
 import { AuthenticatorMachineContext } from '@aws-amplify/ui-react-core';
 
-// TODO extend from ViewProps, add button style related style props
 export interface FederatedProviderButtonsProps
   extends Pick<
-    AuthenticatorMachineContext,
-    'route' | 'socialProviders' | 'toFederatedSignIn'
-  > {}
+      AuthenticatorMachineContext,
+      'route' | 'socialProviders' | 'toFederatedSignIn'
+    >,
+    ViewProps {
+  /*
+   * @description
+   * Styles for the buttons
+   */
+  buttonStyle?: StyleProp<ViewStyle>;
+
+  /*
+   * @description
+   * Styles for the divider text
+   */
+  textStyle?: StyleProp<TextStyle>;
+
+  /*
+   * @description
+   * Styles for the divider line
+   */
+  dividerStyle?: StyleProp<ViewStyle>;
```

Comment:
The [`Divider`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react-native/src/primitives/Divider/types.ts#L3) itself exposes `labelStyle` and `lineStyle` as well as a `style` prop that us applied to the containing `View`. Any reason not to extend from the existing types and follow the naming patterns?